pkg/core/resources/apis/core/interfaces.go

```diff
@@ -4,4 +4,5 @@ type Port interface {
 	GetName() string
 	GetValue() uint32
 	GetNameOrStringifyPort() string
+	GetProtocol() string
```

Comment:
I know we're very bad at documenting interfaces but can we try to improve?

for example: this GetProtocol does it returned a sanitized version of the protocol?

Reply:
added docs